plutus-contract/src/Language/Plutus/Contract/Effects/RPC.hs

```diff
@@ -83,17 +88,37 @@ type HasRPCClient r s = HasEndpoint (RPCResponseEndpoint r) (Either (RPCError r)
 type RPCServer r = Endpoint (RPCRequestEndpoint r) (RPCParams (RPCRequest r))
 type HasRPCServer r s = HasEndpoint (RPCRequestEndpoint r) (RPCParams (RPCRequest r)) s
 
+-- | How many times an RPC call should be retried if the target contract instance is busy.
+data Retries = NoRetries | MaxRetries Natural
+
+-- | Call an endpoint on another contract instance.
 callRPC :: forall r s.
     ( HasOwnId s
+    , HasAwaitSlot s
     , HasEndpoint (RPCResponseEndpoint r) (Either (RPCError r) (RPCResponse r)) s
     , HasContractNotify s
     , RPC r
     , KnownSymbol (RPCRequestEndpoint r)
     )
-    => ContractInstanceId
+    => Retries
+    -> ContractInstanceId
     -> RPCRequest r
     -> Contract s RPCCallError (Either (RPCError r) (RPCResponse r))
-callRPC = call @(RPCRequestEndpoint r) @(RPCResponseEndpoint r) @(RPCRequest r)  @(RPCResponse r) @(RPCError r)
+callRPC retries targetInstance requestArgs =
+    let inner :: Contract s RPCCallError (Either (RPCError r) (RPCResponse r))
+        inner = call @(RPCRequestEndpoint r) @(RPCResponseEndpoint r) @(RPCRequest r)  @(RPCResponse r) @(RPCError r) targetInstance requestArgs
+        maxRetries = case retries of { NoRetries -> 0; MaxRetries n -> n }
+
+        go :: Natural -> Contract s RPCCallError (Either (RPCError r) (RPCResponse r))
+        go i = do
+            rpcResult <- mapError absurd $ runError inner
+            case rpcResult of
+                Left (e@(RPCCallError (EndpointNotAvailable _ _))) -> do
+                    unless (i < maxRetries) (throwError e)
+                    mapError RPCOtherError (waitNSlots 2) >> go (i + 1)
```

Comment:
This is a linear backoff strategy. Normally people use an exponential one. Either way we should document what the retry policies mean!

Reply:
This was just me being too lazy to do an exponential one, but I'll change it :)

Reply:
If the documentation was "this is linear but we should make it exponential if it ever matters" that would be fine :p